api/apps/v1alpha1/nemo_guardrails_service_types.go

```diff
@@ -0,0 +1,746 @@
+/*
+Copyright 2024.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"fmt"
+	"maps"
+	"os"
+
+	rendertypes "github.com/NVIDIA/k8s-nim-operator/internal/render/types"
+	utils "github.com/NVIDIA/k8s-nim-operator/internal/utils"
+	monitoringv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	autoscalingv2 "k8s.io/api/autoscaling/v2"
+	corev1 "k8s.io/api/core/v1"
+	networkingv1 "k8s.io/api/networking/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+const (
+	// NemoGuardrailServiceConditionReady indicates that the NEMO GuardrailService is ready.
+	NemoGuardrailServiceConditionReady = "Ready"
+	// NemoGuardrailServiceConditionFailed indicates that the NEMO GuardrailService has failed.
+	NemoGuardrailServiceConditionFailed = "Failed"
+
+	// NemoGuardrailServiceStatusPending indicates that NEMO GuardrailService is in pending state
+	NemoGuardrailServiceStatusPending = "Pending"
+	// NemoGuardrailServiceStatusNotReady indicates that NEMO GuardrailService is not ready
+	NemoGuardrailServiceStatusNotReady = "NotReady"
+	// NemoGuardrailServiceStatusReady indicates that NEMO GuardrailService is ready
+	NemoGuardrailServiceStatusReady = "Ready"
+	// NemoGuardrailServiceStatusFailed indicates that NEMO GuardrailService has failed
+	NemoGuardrailServiceStatusFailed = "Failed"
+)
+
+// NemoGuardrailServiceSpec defines the desired state of NemoGuardrailService
+type NemoGuardrailServiceSpec struct {
+	Image   Image           `json:"image,omitempty"`
+	Command []string        `json:"command,omitempty"`
+	Args    []string        `json:"args,omitempty"`
+	Env     []corev1.EnvVar `json:"env,omitempty"`
+	// The name of an existing pull secret containing the NGC_API_KEY
+	AuthSecret string `json:"authSecret"`
```

Comment:
is `AuthSecret` a required param for Guardrails? It is not pulling any models from NGC, but is this required for an API access from the API catalog endpoint?

Reply:
Exactly, that is why I leave this field